pkg/users/accounts-list.js

```diff
@@ -239,14 +255,39 @@ const mapGroupsToAccount = (accounts, groups) => {
     });
 };
 
-const GroupsList = ({ groups, accounts }) => {
-    const [isExpanded, setIsExpanded] = useState(false);
+const GroupsList = ({ groups, accounts, isExpanded, setIsExpanded }) => {
+    const { options } = usePageLocation();
+    const [currentTextFilter, setCurrentTextFilter] = useState(options.group || '');
     const columns = [
         { title: _("Group name"), sortable: true },
         { title: _("ID"), sortable: true },
         { title: _("# of users"), sortable: true },
         { title: _("Accounts") },
     ];
+    const filtered_groups = groups.filter(group => {
+        if (currentTextFilter !== "" &&
+            (group.name.toLowerCase().indexOf(currentTextFilter.toLowerCase()) === -1) &&
+            (group.gid.toString().indexOf(currentTextFilter.toLowerCase()) === -1))
+            return false;
+
+        return true;
+    });
+    const ref = useRef();
+
+    useEffect(() => {
+        if (ref.current != currentTextFilter) {
+            ref.current = currentTextFilter;
+            const newOptions = { ...options };
+            if (currentTextFilter) {
+                newOptions.group = currentTextFilter;
+            } else {
+                delete newOptions.group;
+            }
+            cockpit.location.go([], newOptions);
+        } else {
+            setCurrentTextFilter(options.group || "");
+        }
+    }, [currentTextFilter, setCurrentTextFilter, options]);
```

Comment:
This pattern is rather dangerous. It's easy to get into an endless loop. Also, the setter should surely not be part of the dependencies.